api/shape/nth.js

```diff
@@ -1,35 +1,33 @@
-import './each.js';
-
 import Empty from './Empty.js';
 import Group from './Group.js';
 import Shape from './Shape.js';
+import { each } from './each.js';
 
-const eachOp = Shape.ops.get('each');
-
-export const nth = Shape.registerMethod(
+export const nth = Shape.registerMethod2(
   ['nth', 'n'],
-  (...ns) =>
-    async (shape) => {
-      const candidates = await eachOp(
-        (leaf) => leaf,
-        (...leafs) =>
-          (shape) =>
-            leafs
-      )(shape);
-      const group = [];
-      for (let nth of ns) {
-        if (nth < 0) {
-          nth = candidates.length - nth;
-        }
-        const candidate = candidates[nth];
-        if (candidate === undefined) {
-          group.push(Empty());
-        } else {
-          group.push(candidate);
-        }
+  ['input', 'numbers'],
```

Comment:
How do these work? Is this for documentation?

Reply:
This tells it how to do the magical decoding stuff.

This means, "look for an input shape first, then look for numbers".

But, yeah, it should also be used to generate documentation and UI automatically.

These then get passed in with the corresponding arguments to the actual function (which should move out of here and into geometry)